File: src/main/java/cptn/util/XException.java
package cptn.util;

/**
 * 自定义异常处理
 * @author cptn
 *
 */
public class XException extends RuntimeException {

	private static final long serialVersionUID = 1L;
	private int errCode = 0;
	
	public XException() {
		super();
	}

	public XException(String msg) {
		super(msg);
	}
	
	public XException(int err, String msg) {
		super(msg);
		errCode = err;
	}

	public int getErrCode() {
		return errCode;
	}
	
	@Override
	public String toString() {
		return errCode + ": " + getMessage();
	}
}

File: src/main/java/cptn/util/XException.kt
package cptn.util

// 自定义异常
class XException : RuntimeException {
    var errCode: Int = 0

    constructor() : super() {}

    constructor(msg: String) : super(msg) {}

    constructor(err: Int, msg: String) : super(msg) {
        errCode = err
    }

    override fun toString(): String {
        return "$errCode: $message"
    }
}
